qcodes/dataset/guid_helpers.py

```diff
@@ -77,16 +78,30 @@ def guids_from_list_str(s: str) -> Optional[Tuple[str, ...]]:
         ('07fd7195-c51e-44d6-a085-fa8274cf00d6',
         '070d7195-c51e-44d6-a085-fa8274cf00d6')
     """
-    guid = r"[\da-f]{8}-(?:[\da-f]{4}-){3}[\da-f]{12}"
-    captured_guid = fr"(?:\s*['\"]?({guid})['\"]?\s*)"
-    open_parens = r"[\[\(\{]"
-    close_parens = r"[\]\)\}]"
-    m = re.match(
-        fr"^\s*{open_parens}?"
-        fr"(?:{captured_guid},)*{captured_guid}?"
-        fr"{close_parens}?\s*$",
-        s,
-    )
-    if m is None:
+    if s == "":
+        return tuple()
+
+    try:
+        parsed_expression = ast.parse(s, mode='eval')
+    except SyntaxError:
+        return None
+
+    if not hasattr(parsed_expression, 'body'):
         return None
-    return tuple(v for v in m.groups() if v is not None)
+
+    parsed = cast(ast.Expression, parsed_expression).body
+
+    if isinstance(parsed, ast.Str):
+        if len(parsed.s) > 0:
+            return (parsed.s,)
+        else:
+            return tuple()
+
+    if not isinstance(parsed, (ast.List, ast.Tuple, ast.Set)):
+        return None
+
+    if not all(isinstance(e, ast.Constant) for e in parsed.elts):
```

Comment:
Looks like the current code fails since this is not a list of ast.Constant but a list of ast.Str which have a .s rather than a .v member

Reply:
indeed! thanks!